packages/vite/src/node/plugins/css.ts

```diff
@@ -498,14 +503,34 @@ export function cssPostPlugin(config: ResolvedConfig): Plugin {
           opts.format === 'cjs' ||
           opts.format === 'system'
         ) {
-          const cssAssetName = chunk.name + '.css'
+          const cssAssetName = ensureFileExt(
+            chunk.facadeModuleId
+              ? normalizePath(path.relative(config.root, chunk.facadeModuleId))
+              : chunk.name,
+            '.css'
+          )
 
           chunkCSS = resolveAssetUrlsInCss(chunkCSS, cssAssetName)
           chunkCSS = await finalizeCss(chunkCSS, true, config)
 
+          const output = config.build?.rollupOptions?.output
+          const assetFileNames =
+            (output && !Array.isArray(output)
+              ? output.assetFileNames
+              : undefined) ??
+            // defaults to '<assetsDir>/[name].[hash][extname]'
+            // slightly different from rollup's one ('assets/[name]-[hash][extname]')
+            path.posix.join(config.build.assetsDir, '[name].[hash][extname]')
+
```

Comment:
This is copied verbatim from `asset.ts` so we could potentially extract it to function if we wanted to keep that logic and naming pattern centralised.

Reply:
Done, there was also a fix for plugin legacy with custom asset file names. So this abstraction was needed for it